Ds3/Helpers/Streams/StreamTranslator.cs

```diff
@@ -125,7 +126,18 @@ public override void Write(byte[] buffer, int offset, int count)
                 this._streamStore.Access(range.Context, stream =>
                 {
                     stream.Position = range.Range.Start;
-                    stream.Write(buffer, progress + offset, rangeLength);
+                    try
+                    {
+                        stream.Write(buffer, progress + offset, rangeLength);
+                    }
+                    catch (IOException e)
+                    {
+                        if (e.Message.StartsWith("There is not enough space on the disk"))
```

Comment:
Is there another way that we can detect this?  The string comparison seems brittle.